src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java

```diff
@@ -130,6 +130,10 @@ public void removeModelWaitingForAssessment(long exerciseId, long modelSubmissio
             return new ArrayList<>();
         }
 
+        // Print statistics of the current modeling exercise for internal analysis. Disabled by default due to performance reasons.
+        // compassCalculationEngines.get(exerciseId)
```

Comment:
It would be nice if this could be done through a log level :thinking: Maybe introduce a compass log level as a setting?

Reply:
After discussing this with @krusche we decided to add a new admin endpoint that triggers the calculation and print of the statistics. In the future, the endpoint should also return the statistics to the client as JSON. I removed the call to `printStatistic` here and added a separate method for it in the `CompassService`.